modules/market/src/main/java/com/opengamma/strata/market/surface/DefaultSurfaceMetadata.java

```diff
@@ -122,10 +133,43 @@ private static void applyDefaults(Builder builder) {
 
   //-------------------------------------------------------------------------
   @Override
-  public DefaultSurfaceMetadata withParameterMetadata(List<SurfaceParameterMetadata> parameterMetadata) {
+  public <T> T getInfo(SurfaceInfoType<T> type) {
+    // overridden for performance
+    @SuppressWarnings("unchecked")
+    T value = (T) info.get(type);
+    if (value == null) {
+      throw new IllegalArgumentException(Messages.format("Surface info not found for type '{}'", type));
+    }
+    return value;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T> Optional<T> findInfo(SurfaceInfoType<T> type) {
+    return Optional.ofNullable((T) info.get(type));
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public DefaultSurfaceMetadata withInfo(Map<SurfaceInfoType<?>, Object> additionalInfo) {
+    return toBuilder().addInfo(additionalInfo).build();
+  }
+
+  @Override
+  public DefaultSurfaceMetadata withParameterMetadata(List<? extends SurfaceParameterMetadata> parameterMetadata) {
     return toBuilder().parameterMetadata(parameterMetadata).build();
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Returns a builder that allows this bean to be mutated.
```

Comment:
This is the standard wording from Joda-Beans used everywhere. I've changed it here.